homeassistant/components/trend/binary_sensor.py

```diff
@@ -96,21 +103,50 @@ async def async_setup_platform(
         max_samples = device_config[CONF_MAX_SAMPLES]
         min_gradient = device_config[CONF_MIN_GRADIENT]
         sample_duration = device_config[CONF_SAMPLE_DURATION]
+        min_samples = device_config[CONF_MIN_SAMPLES]
 
-        sensors.append(
-            SensorTrend(
-                hass,
-                device_id,
-                friendly_name,
-                entity_id,
-                attribute,
-                device_class,
-                invert,
+        # check if min_samples is smaller than max_samples
+        issue = f"min_samples_larger_max_samples_{device_id}"
+        if min_samples > max_samples:
+            _LOGGER.error(
+                "Configured min_sample (%d) is larger then max_samples (%d), skip binary sensor %s",
+                min_samples,
                 max_samples,
-                min_gradient,
-                sample_duration,
+                device_id,
             )
-        )
+
+            async_create_issue(
+                hass,
+                DOMAIN,
+                issue,
+                is_fixable=False,
+                severity=IssueSeverity.WARNING,
+                translation_key="min_samples_larger_max_samples",
+                translation_placeholders={
+                    "device_id": device_id,
```

Comment:
`device_id` is really a terrible name, it should be `trend_sensor` or something like that. that should be fixed in a follow up PR though

Reply:
Will do